pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go

```diff
@@ -83,10 +85,43 @@ func (s *preFilterState) Clone() framework.StateData {
 // failed by this plugin schedulable.
 func (pl *NodeAffinity) EventsToRegister() []framework.ClusterEventWithHint {
 	return []framework.ClusterEventWithHint{
-		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterNodeChange},
 	}
 }
 
+// isSchedulableAfterNodeChange is invoked whenever a node changed. It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *NodeAffinity) isSchedulableAfterNodeChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
```

Comment:
@ahg-g @alculquicondor 

Another topic. Does it make sense to have `error` in the returning value? When the hint fn returns `error`, the scheduling queue logs the error and treats it as QueueAfterBackoff.

I'm saying this because it's probably gonna be a common mistake that when something unexpected (probably due to bugs) happens, just log the error and **return** **QueueSkip**. In such unexpected error cases, generally speaking, we should treat the event as QueueAfterBackoff so that we can prevent the bug from resulting in the Pod stuck in the unschedulable pod pool

Reply:
@sanposhiho @alculquicondor I raise a pr, please review https://github.com/kubernetes/kubernetes/pull/119290